### PR TITLE
implement widget injured accidents with pedestrian

### PR DESCRIPTION
**Pull request by 9Syncopation**
TODO implement the legendary on top of the bar chart

### CODE REVIEW

File: src/components/molecules/StackedBarChartView.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  xLabel2: string;
+  xLabel3: string;
+  yLabel: string | number;
+  textLabel?: string;
+}
+
+const StackedBarChartView: FC<IProps> = ({ data, xLabel1, xLabel2, xLabel3 }) => {
+  return (
+    <ResponsiveContainer>
+      <BarChart data={data} margin={{ top: 20, bottom: 15 }}>
```
Comment (yuvalbl) [inline src/components/molecules/StackedBarChartView.tsx:17]: theme spacing / Box

File: src/models/WidgetData.ts [hunk 2]
```diff
@@ -156,3 +156,11 @@ export interface IWidgetAccidentCountByCarType extends IWidgetDataBase {
     percentage_country: number;
   }[];
 }
+export interface IWidgetInjuredAccidentsWithPedestrians extends IWidgetDataBase {
+  items: {
+    YEAR: number;
+    KILLED_INJURY_SEVERITY_COUNT: string | number;
```
Comment (yuvalbl) [inline src/models/WidgetData.ts:162]: pass data as is - doe conversions inside the component if required

File: src/components/molecules/StackedBarChartView.tsx (hunk 1 above)
Comment (9Syncopation) [reply src/components/molecules/StackedBarChartView.tsx]: removed the styling

File: src/models/WidgetData.ts (hunk 2 above)
Comment (9Syncopation) [reply src/models/WidgetData.ts]: done

**Comment by 9Syncopation** [comment]
add the Legend to the chart.
please advise if you think of a more elegant way to import the legend naming (name1, name2...)

File: src/components/molecules/StackedBarChartView.tsx [hunk 3]
```diff
@@ -9,16 +9,20 @@ interface IProps {
...
   return (
     <ResponsiveContainer>
-      <BarChart data={data} margin={{ top: 20, bottom: 15 }}>
+      <BarChart data={data}>
+        <Legend layout="vertical" align="right" wrapperStyle={{ top: 5, right: 0, textAlign: 'right' }}></Legend>
         <XAxis dataKey="year" axisLine={false} tickLine={false} />
-        <Bar dataKey={xLabel1} stackId="a" fill={roseColor} />
-        <Bar dataKey={xLabel2} stackId="a" fill={honeyColor} />
-        <Bar dataKey={xLabel3} stackId="a" fill={yellowColor} />
+        <Bar dataKey={xLabel1} name={name1} stackId="a" fill={roseColor} />
```
Comment (yuvalbl) [inline src/components/molecules/StackedBarChartView.tsx:23]: replace label1-3, name1-3:  with `bar` (array of objects)

```
const colors = [color1, color2]
bars.map((bar, index) => (
  <Bar dataKey={bar.dataKey} name={bar.name} stackId="stacked" fill={colors[index]} />
))
```

**Review by yuvalbl — APPROVED** [review]

**Pull request merged**